src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -137,31 +162,45 @@ public void Enable()
             _masterGpioController.Write((int)_reset, PinValue.High);
         }
 
+        /// <summary>
+        /// Reads a byte from a register.
+        /// </summary>
+        /// <param name="registerAddress">The register address to read.</param>
+        /// <param name="port">The I/O port used with the registers.</param>
+        /// <param name="bank">The bank type that determines how the register is mapped.</param>
+        /// <returns>The data read from the register.</returns>
         public byte Read(Register.Address registerAddress, Port port = Port.PortA, Bank bank = Bank.Bank1)
         {
             byte[] data = Read(registerAddress, 1, port, bank);
             return data[0];
         }
 
+        /// <summary>
+        /// Reads a bit from a register.
+        /// </summary>
+        /// <param name="registerAddress">The register address to read.</param>
+        /// <param name="bitNumber">The register bit number to read.</param>
+        /// <param name="port">The I/O port used with the registers.</param>
+        /// <param name="bank">The bank type that determines how the register is mapped.</param>
+        /// <returns>The value of the register bit read.</returns>
         public bool ReadBit(Register.Address registerAddress, int bitNumber, Port port = Port.PortA, Bank bank = Bank.Bank1)
         {
-            ValidateBitNumber(bitNumber);
```

Comment:
I see that you were effectively performing this check twice so it makes sense to remove it, but should we instead keep the Validation here instead and remove it from GetBit so that we don't perform a Read if the Bit is out of range?

Reply:
I was thinking the same, but thought the validation would've been better to keep just in GitBit so it catches in both.  I can add back if you think there is no harm in this case.

Reply:
Will you ever call GetBit through a different path where we can't do validation sooner? If not, then I would rather do the validation soon in order to avoid the Read when we know the bit won't be valid.

Reply:
Added back